**Apply whichever tests are applicable on Attrition Dataset and make inference from it**

In [1]:
import pandas as pd
dataset = pd.read_csv("general_data.csv")
dataset.columns
dataset.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

**1. MannnWhitney test**

**Attrition and DistanceFromHome**

In [3]:
from sklearn import preprocessing
from scipy.stats import pearsonr
dataset["Attrition"].replace(to_replace=("No","Yes"),value=(0,1),inplace=True)
dataset.head(5)

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
1   31          1  Travel_Frequently  Research & Development   
2   32          0  Travel_Frequently  Research & Development   
3   38          0         Non-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
1                10          1  Life Sciences              1           2   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   

   Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0  Female  ...                 1.0      Y                11              8   
1  Female  ...                 0.0      Y                23              8   
2    Male  ...                 1.0      Y                15              8   
3    Male  ...                 3.0      Y                11              8   
4    Male  ...                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [5]:
a1=dataset[dataset["Attrition"]==0]
a1.head(5)

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
2   32          0  Travel_Frequently  Research & Development   
3   38          0         Non-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   
5   46          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   
5                 8          3  Life Sciences              1           6   

   Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0  Female  ...                 1.0      Y                11              8   
2    Male  ...                 1.0      Y                15              8   
3    Male  ...                 3.0      Y                11              8   
4    Male  ...                 4.0      Y                12              8   
5  Female  ...                 3.0      Y                13              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   
5                 0              28.0                      5               7   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
2                        0                     3  
3                        7                     5  
4                        0                     4  
5                        7                     7  

[5 rows x 24 columns]

In [6]:
a2=dataset[dataset["Attrition"]==1]
a2.head(5)

Age  Attrition     BusinessTravel              Department  \
1    31          1  Travel_Frequently  Research & Development   
6    28          1      Travel_Rarely  Research & Development   
13   47          1         Non-Travel  Research & Development   
28   44          1  Travel_Frequently  Research & Development   
30   26          1      Travel_Rarely  Research & Development   

    DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
1                 10          1  Life Sciences              1           2   
6                 11          2        Medical              1           7   
13                 1          1        Medical              1          14   
28                 1          2        Medical              1          29   
30                 4          3        Medical              1          31   

    Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
1   Female  ...                 0.0      Y                23              8   
6     Male  ...                 2.0      Y                20              8   
13    Male  ...                 1.0      Y                11              8   
28    Male  ...                 3.0      Y                14              8   
30    Male  ...                 2.0      Y                11              8   

    StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
1                  1               6.0                      3               5   
6                  1               5.0                      2               0   
13                 2              10.0                      4              10   
28                 1              19.0                      2               1   
30                 0               5.0                      5               3   

    YearsSinceLastPromotion  YearsWithCurrManager  
1                         1                     4  
6                         0                     0  
13                        9                     9  
28                        0                     0  
30                        0                     2  

[5 rows x 24 columns]

In [9]:
from scipy.stats import mannwhitneyu
stats,p=mannwhitneyu(a1.DistanceFromHome,a2.DistanceFromHome)
print(stats,p)

1312110.0 0.4629185205822659


As the P value of 0.4629185205822659 is > 0.05, the H0 is accepted and Ha is rejected.
H0: There is no significant differences in the Distance From Home between attrition (Y) and attirition (N)
Ha: There is significant differences in the Distance From Home between attrition (Y) and attirition (N)

**Attrition and MonthlyIncome**

In [11]:
stats,p=mannwhitneyu(a1.MonthlyIncome,a2.MonthlyIncome)
print(stats,p)

1264900.5 0.053577283839938566


As the P value of 0.053577283839938566 is > 0.05, the H0 is accepted and Ha is accepted. H0: There is no significant differences in the MonthlyIncome between attrition (Y) and attirition (N) Ha: There is significant differences in the MonthlyIncome between attrition (Y) and attirition (N)

 **Attrition and TotalWorkingYears**

In [12]:
stats,p=mannwhitneyu(a1.TotalWorkingYears,a2.TotalWorkingYears)
print(stats,p)

909958.0 2.894653260702919e-39


As the P value is 2.894653260702919e-39, which is < than 0.05, the H0 is rejected and ha is accepted. H0: There is no significant differences in the Total Working Years between attrition (Y) and attirition (N) Ha: There is significant differences in the Total Working Years between attrition (Y) and attirition (N)

**Attrition and YearsAtCompany**

In [13]:
stats,p=mannwhitneyu(a1.YearsAtCompany,a2.YearsAtCompany)
print(stats,p)

923238.0 6.047598261693028e-37


As the P value is 6.047598261693028e-37, which is < than 0.05, the H0 is rejected and ha is accepted.
H0: There is no significant differences in the YearsAtCompany between attrition (Y) and attirition (N)
Ha: There is significant differences in the YearsAtCompany between attrition (Y) and attirition (N)

**2. Kruskal wallis test**

**Attrition and Department**

In [14]:
dataset.Department.unique()

array(['Sales', 'Research & Development', 'Human Resources'], dtype=object)

In [17]:
dataset["Department"].replace(to_replace=("Sales","Research & Development","Human Resources"),value=(0,1,2),inplace=True)
dataset.head(5)

Age  Attrition     BusinessTravel  Department  DistanceFromHome  Education  \
0   51          0      Travel_Rarely           0                 6          2   
1   31          1  Travel_Frequently           1                10          1   
2   32          0  Travel_Frequently           1                17          4   
3   38          0         Non-Travel           1                 2          5   
4   32          0      Travel_Rarely           1                10          1   

  EducationField  EmployeeCount  EmployeeID  Gender  ...  NumCompaniesWorked  \
0  Life Sciences              1           1  Female  ...                 1.0   
1  Life Sciences              1           2  Female  ...                 0.0   
2          Other              1           3    Male  ...                 1.0   
3  Life Sciences              1           4    Male  ...                 3.0   
4        Medical              1           5    Male  ...                 4.0   

  Over18 PercentSalaryHike  StandardHours  StockOptionLevel TotalWorkingYears  \
0      Y                11              8                 0               1.0   
1      Y                23              8                 1               6.0   
2      Y                15              8                 3               5.0   
3      Y                11              8                 3              13.0   
4      Y                12              8                 2               9.0   

   TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
0                      6               1                        0   
1                      3               5                        1   
2                      2               5                        0   
3                      5               8                        7   
4                      2               6                        0   

   YearsWithCurrManager  
0                     0  
1                     4  
2                     3  
3                     5  
4                     4  

[5 rows x 24 columns]

In [19]:
b1=dataset[dataset["Department"]==0]
b1.head(5)

Age  Attrition BusinessTravel  Department  DistanceFromHome  Education  \
0    51          0  Travel_Rarely           0                 6          2   
18   35          0  Travel_Rarely           0                 7          4   
21   50          0  Travel_Rarely           0                 8          4   
27   37          0  Travel_Rarely           0                 5          1   
29   38          0  Travel_Rarely           0                 2          3   

   EducationField  EmployeeCount  EmployeeID  Gender  ...  NumCompaniesWorked  \
0   Life Sciences              1           1  Female  ...                 1.0   
18  Life Sciences              1          19    Male  ...                 7.0   
21  Life Sciences              1          22    Male  ...                 3.0   
27      Marketing              1          28    Male  ...                 1.0   
29      Marketing              1          30  Female  ...                 9.0   

   Over18 PercentSalaryHike  StandardHours  StockOptionLevel  \
0       Y                11              8                 0   
18      Y                16              8                 0   
21      Y                23              8                 0   
27      Y                11              8                 0   
29      Y                12              8                 0   

   TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                1.0                      6               1   
18              10.0                      5               7   
21              28.0                      2              10   
27              17.0                      2              17   
29              10.0                      3               2   

    YearsSinceLastPromotion  YearsWithCurrManager  
0                         0                     0  
18                        6                     2  
21                        1                     6  
27                        5                     7  
29                        1                     2  

[5 rows x 24 columns]

In [20]:
b2=dataset[dataset["Department"]==1]
b2.head(5)

Age  Attrition     BusinessTravel  Department  DistanceFromHome  Education  \
1   31          1  Travel_Frequently           1                10          1   
2   32          0  Travel_Frequently           1                17          4   
3   38          0         Non-Travel           1                 2          5   
4   32          0      Travel_Rarely           1                10          1   
5   46          0      Travel_Rarely           1                 8          3   

  EducationField  EmployeeCount  EmployeeID  Gender  ...  NumCompaniesWorked  \
1  Life Sciences              1           2  Female  ...                 0.0   
2          Other              1           3    Male  ...                 1.0   
3  Life Sciences              1           4    Male  ...                 3.0   
4        Medical              1           5    Male  ...                 4.0   
5  Life Sciences              1           6  Female  ...                 3.0   

  Over18 PercentSalaryHike  StandardHours  StockOptionLevel TotalWorkingYears  \
1      Y                23              8                 1               6.0   
2      Y                15              8                 3               5.0   
3      Y                11              8                 3              13.0   
4      Y                12              8                 2               9.0   
5      Y                13              8                 0              28.0   

   TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
1                      3               5                        1   
2                      2               5                        0   
3                      5               8                        7   
4                      2               6                        0   
5                      5               7                        7   

   YearsWithCurrManager  
1                     4  
2                     3  
3                     5  
4                     4  
5                     7  

[5 rows x 24 columns]

In [21]:
b3=dataset[dataset["Department"]==2]
b3.head(5)

Age  Attrition BusinessTravel  Department  DistanceFromHome  Education  \
79    29          0  Travel_Rarely           2                 8          3   
100   36          0     Non-Travel           2                28          3   
105   33          1  Travel_Rarely           2                28          2   
112   21          1  Travel_Rarely           2                10          2   
134   54          0  Travel_Rarely           2                 9          3   

      EducationField  EmployeeCount  EmployeeID  Gender  ...  \
79           Medical              1          80    Male  ...   
100  Human Resources              1         101    Male  ...   
105  Human Resources              1         106  Female  ...   
112  Human Resources              1         113    Male  ...   
134    Life Sciences              1         135  Female  ...   

     NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
79                  0.0      Y                22              8   
100                 4.0      Y                22              8   
105                 1.0      Y                21              8   
112                 1.0      Y                12              8   
134                 4.0      Y                23              8   

     StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
79                  2               4.0                      3   
100                 0               7.0                      4   
105                 1               1.0                      3   
112                 1               1.0                      2   
134                 1              24.0                      2   

     YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
79                3                        2                     2  
100               1                        0                     0  
105               1                        0                     0  
112               1                        0                     0  
134               4                        1                     2  

[5 rows x 24 columns]

In [34]:
from scipy.stats import kruskal
stats,p=kruskal(b1.Attrition,b2.Attrition,b3.Attrition)
print(stats,p)

29.083678490265882 4.836814804414598e-07


As the P value is 4.836814804414598e-07, which is < than 0.05, the H0 is rejected and ha is accepted.
H0: There is no significant differences in the various departments and attirition 
Ha: There is significant differences in the various departments and attrition

**Statistical test- Seperate t test**

**Attrition and Distance From Home**

In [35]:
from scipy.stats import ttest_ind
stats,p=ttest_ind(a1.DistanceFromHome,a2.DistanceFromHome)
print(stats,p)

0.6460416038042738 0.518286042805572


As the P value is 0.518286042805572, which is > than 0.05, the H0 is accepted and ha is rejected.
H0: There is no significant differences in the Distance From Home between attrition (Y) and attirition (N)
Ha: There is significant differences in the Distance From Home between attrition (Y) and attirition (N)

**Attrition and  MonthlyIncome**

In [36]:
stats,p=ttest_ind(a1.MonthlyIncome,a2.MonthlyIncome)
print(stats,p)

2.0708863763619316 0.03842748490605113


As the P value is 0.03842748490605113, which is < than 0.05, the H0 is rejected and ha is accepted. H0: There is no significant differences in the Monthly Income between attrition (Y) and attirition (N) Ha: There is significant differences in the Monthly Income between attrition (Y) and attirition (N)

**Attrition and YearsAtCompany**

In [38]:
stats,p=ttest_ind(a1.YearsAtCompany,a2.YearsAtCompany)
print(stats,p) 

9.004357011787226 3.163883122491456e-19


As the P value is 3.163883122491456e-19, which is < than 0.05, the H0 is rejected and ha is accepted.
H0: There is no significant differences in the Years At Company between attrition (Y) and attirition (N)
Ha: There is significant differences in the Years At Company between attrition (Y) and attirition (N)